src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -5218,7 +5218,7 @@ private enum NameOptions
         /// <returns></returns>
         private bool IsTrueIdentifier()
         {
-            if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken)
+            if (this.CurrentToken.Kind == SyntaxKind.IdentifierToken && token.Text is not ("nint" or "nuint"))
```

Comment:
:warning: warning: Most likely, this will produce an error if "nint" is used as a variable name, which is certainly not desired. I'm trying to find a solution, but unfortunately, until now, I'm not able to.

I'm also not sure what other cases this will affect, and if these effects are desired or not.